### PR TITLE
[UI v2] feat: Adds EmptyState for Task Run Concurrency Limits

### PR DESCRIPTION
**Pull request by devinvillarosa**
1. Adds EmptyState for Task Run Concurrency Limits
2. Adds simple tests for component callback
3. Adds docs entry

### Checklist
- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

Related to https://github.com/PrefectHQ/prefect/issues/15512

### CODE REVIEW

File: ui-v2/src/components/ui/docs-link.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { Button } from "./button";
 const DOCS_LINKS = {
 	"global-concurrency-guide":
 		"https://docs.prefect.io/v3/develop/global-concurrency-limits",
+	"task-concurrency-guide": "https://docs.prefect.io/v3/develop/task-runners",
```
Comment (devinvillarosa) [inline ui-v2/src/components/ui/docs-link.tsx:8]: I'm unsure if this is the correct url

Comment (desertaxle) [reply ui-v2/src/components/ui/docs-link.tsx:8]: This is the link we want: https://docs.prefect.io/v3/develop/task-run-limits

File: ui-v2/src/components/concurrency/task-run-concurrency-limit-empty-state.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	EmptyStateTitle,
+} from "@/components/ui/empty-state";
+import { PlusIcon } from "lucide-react";
+
+type Props = {
+	onClick: () => void;
+};
+export const TaskRunConcurrencyLimitEmptyState = ({ onClick }: Props) => (
+	<EmptyState>
+		<EmptyStateIcon id="CircleArrowOutUpRight" />
```
Comment (desertaxle) [inline ui-v2/src/components/concurrency/task-run-concurrency-limit-empty-state.tsx:17]: Random thought: does defining the icon by ID like this affect tree shaking for `react-lucide`?

Comment (devinvillarosa) [reply ui-v2/src/components/concurrency/task-run-concurrency-limit-empty-state.tsx:17]: I don't think so: https://lucide.dev/guide/packages/lucide-react#one-generic-icon-component 

I think the next followup PR, we can create a list of Icons only available to the ui using its direct imports.

And then we can create a  Record of _ui-v2's_ Icons and export it that way

Comment (desertaxle) [reply ui-v2/src/components/concurrency/task-run-concurrency-limit-empty-state.tsx:17]: Gotcha, that sounds like a good way to abstract icon stuff and make it easy to introduce icons from other sources.

Comment (devinvillarosa) [reply ui-v2/src/components/concurrency/task-run-concurrency-limit-empty-state.tsx:17]: Addressing in this PR: https://github.com/PrefectHQ/prefect/pull/16208

**Review by desertaxle — APPROVED** [review]

**Pull request merged**